c10/core/AllocatorConfig.h

```diff
@@ -0,0 +1,217 @@
+#pragma once
+
+#include <c10/core/DeviceType.h>
+#include <c10/util/Exception.h>
+#include <c10/util/llvmMathExtras.h>
+
+#include <array>
+#include <atomic>
+#include <mutex>
+#include <string>
+#include <vector>
+
+namespace c10::CachingAllocator {
+
+// "large" allocations may be packed in 20 MiB blocks
+const size_t kLargeBuffer = 20971520;
+
+/**
+ * Note [AllocatorConfig design]
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * This class configures the allocator for both device and host memory
+ * management. A single AllocatorConfig is used across all devices, such as CUDA
+ * and XPU, assuming that environment variables apply universally.
+ *
+ * It is designed to *ONLY* contain configuration options that can be set via
+ * environment variables.
+ *
+ * Naming Convention:
+ * - Public API names in AllocatorConfig should be device-generic. For example,
+ *     `use_release_lock_on_cudamalloc` is not ideal; instead, use
+ *     `use_release_lock_on_device_malloc` to maintain neutrality.
+ * - Members prefixed with `pinned_` are specific to the host/pinned allocator.
+ * - Environment variable names should also be device-generic to ensure
+ *     consistency across different hardware backends.
+ *
+ * Environment Variables:
+ * - The default environment variable for configuration is `PYTORCH_ALLOC_CONF`.
+ * - For backward compatibility, `PYTORCH_CUDA_ALLOC_CONF` is also supported
+ *     with lower priority.
+ */
+
+class C10_API AllocatorConfig {
```

Comment:
I feel like this is very much about accelerator devices only? If so, I would suggest updating the name here.

Also update the doc above to be less about the history (cuda vs not) and more about the current state: any generic accelerator uses this.

Reply:
It feels a bit forced to move the current CUDA-specific configurations (the other backend maybe support or not) into the generic `AllocatorConfig`. Instead, I suggest that we keep only the common configurations in `AllocatorConfig` and retain the CUDA-specific ones in `CUDAAllocatorConfig`. Meanwhile, `CUDAAllocatorConfig` will remain internal to `CUDACachingAllocator` and not be exposed beyond it.
And `AllocatorConfig` will be responsible for both device and pin memory allocator. Is this reasonable to you?

Reply:
Renamed the class to `AcceleratorAllocatorConfig` to clarify its scope as a configuration manager for accelerator backends (e.g., CUDA, XPU). The original name `AllocatorConfig` is now reserved for a potential future base class that can unify configuration handling for both CPU and accelerator allocators, should similar requirements arise for the CPU path.